sensor/FlowUploadSensor.js

```diff
@@ -0,0 +1,209 @@
+/*    Copyright 2016 Firewalla LLC
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+'use strict'
+
+let util = require('util')
+let async = require('asyncawait/async')
+let await = require('asyncawait/await')
+let zlib = require('zlib')
+let Promise = require('bluebird')
+
+let log = require('../net2/logger.js')(__filename)
+let Sensor = require('./Sensor.js').Sensor
+let HostManager = require('../net2/HostManager.js')
+let hostManager = new HostManager('cli', 'server')
+let HostTool = require('../net2/HostTool')
+let hostTool = new HostTool()
+let flowTool = require('../net2/FlowTool')()
+let flowUtil = require('../net2/FlowUtil')
+let Bone = require('../lib/Bone.js')
+
+let INTERVAL_MIN = 10 //10 seconds
+let INTERVAL_MAX = 3600 //1 hour
+let INTERVAL_DEFAULT = 900 //15 minutes
+let MAX_UPLOAD_SIZE = 1073741824 //1G
```

Comment:
I assume this is for safety.  If it is, then make the max like 20 megabytes or something.  And if partial, put an indicator somewhere and say, there are total x bytes, we only send y bytes.

And this number should not be bytes either, it should be number of flows.   In case there are a lot of p2p data, we don't want the server to get all that noise if after a certain limit.